h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -838,4 +838,66 @@ public void map(Chunk c, NewChunk nc) {
     }
   }
 
+  /**
+   * shuffle values of a Vec {@link Vec}.
+   * 
+   * The 'new' values of the vector will be the already exisitng one's however randomly shuffeled
+   * 
+   * 
+   */
+  
+  public static class shuffleVec extends MRTask<shuffleVec>{
+    public Vec _vec;
+        
+    public shuffleVec(Vec vec) {this._vec = vec;}
```

Comment:
1. You are calling this MR task with the input vector `vec` which you are saving in every node here. For example, if you run your code with 4 nodes, every node gets its own copy of `vec`. For reading it is no problem, however for writing it is not a good idea (for writing to a vec it is better to use input like `NewChunk nc`).